test/main/assert.js

```diff
@@ -667,18 +667,19 @@ QUnit.test( "rejects", function( assert ) {
 
 ( function() {
 	var previousTestAssert;
+	var firstTestName = "assertions after test finishes throws an error - part 1";
 
 	QUnit.module( "delayed assertions" );
 
-	QUnit.test( "assertions after test finishes throws an error - part 1", function( assert ) {
+	QUnit.test( firstTestName, function( assert ) {
 		assert.expect( 0 );
 		previousTestAssert = assert;
 	} );
 
 	QUnit.test( "assertions after test finishes throws an error - part 2", function( assert ) {
 		assert.expect( 1 );
 		assert.throws( function() {
-			previousTestAssert.ok( true );
-		}, /Assertion occurred after test had finished/ );
+			previousTestAssert.ok( true, "message here" );
+		}, "Assertion message here occurred after test " + firstTestName + " had finished" );
```

Comment:
This might read better with a few more delimiters, maybe something like
```
Assertion 'message here' occurred after test 'my test' had finished.
```
?

Reply:
Agreed. Although test names themselves can also contain quotes at times. I think it would be okay to use multiple lines here. That's somewhat unusual for exception messages, but should be okay here.

For example:

```
Error: Assertion occurred after test finished.
> Test: assertions after test finish cause an error
> Assertion: message here

Reply:
Sounds great to me, will make the change and rebase as I saw that this file just changed.